### DISCUSSION TITLE
Scrolling issue in multi line list

### DISCUSSION THREAD

**Discussion by refaelsh** [jtdaugherty/brick#501] [Q&A] [answered]
Here is how to reproduce:
In the list demo program, in the `listDrawElement :: (Show a) => Bool -> a -> Widget ()` function, change the last line to this:
```haskell
in vBox $ [str "Item " <+> (selStr $ show a), str "Item " <+> (selStr $ show a)]
```
Then add (using the `+` key) 25 items or more.
Go to the last item in the list, e.g., press the down arrow many times.
Now press the up arrow once, and you will see the list scrolling - but it should not scroll.

Please help :-)

**Comment by jtdaugherty** [comment]
Did you also adjust the list item height from `1` to `2`?

**Reply by jtdaugherty** [reply, marked as answer]
Specifically, the last argument to `list` must be `2` instead of `1` in
```haskell
initialState = L.list () (Vec.fromList ['a','b','c']) 1
```

**Reply by refaelsh** [reply]
No. But now I did it and the problem went away. All is fine. Thank you! :-)
Follow-up question, what if my list contains items that are sometime having height of 2, and sometimes height of 3. What should I set the item height in such a case?

**Reply by jtdaugherty** [reply]
The built-in list has a requirement that all items have the same height, precisely so that the scrolling arithmetic works out correctly. So you can't put items in a list that have varying heights; if you do, you'll get exactly the behavior you ran into.

**Reply by refaelsh** [reply]
Yes. I understand. Thank you.
